main.py

```diff
@@ -36,45 +34,41 @@ def getRealTimeStamp():
 SLEEP_TIME = 0.001
 
 
-def main(onCap, checkNetworkClose):
-    finished = False
-    while not finished:
-        # outer loop waits for the window to exists
-        frame_start = time.time()
-        frame_end = frame_start + RATE
-        hwnd = getWindow()
+def main(on_cap, check_network_close):
+    strategy = Strategy()
+
+    # The outer loop makes sure that the program retries constantly even when
+    # capturing device is having trouble
+    while True:
+        try:
+            ts, image = capture.get_image()
 
-        if not hwnd:
-            while time.time() < frame_end:
-                time.sleep(SLEEP_TIME)
+            if not ts and not image:
+                if config["debug.print_packet"]:
+                    print("clean exit")
+
+                break
+        except Exception:
+            time.sleep(RATE)
             continue
 
-        strat = Strategy(hwnd)
-        while checkWindow(hwnd):
-            # inner loop gets fresh data for just the desired window
-            frame_start = time.time()
-            frame_end = frame_start + RATE
-            strat.update(getTimeStamp())
-            result = strat.to_dict()
-            processing_time = time.time() - frame_start
-            # if processing_time > BIG_BOI:
+        frame_end_ts = ts + RATE
+
+        strategy.update(ts, image)
+        result = strategy.to_dict()
+
+        if config["debug.print_packet"]:
+            processing_time = time.time() - ts
             print(processing_time)
-            #    BIG_BOI = processing_time
             print(result)
-            # onCap(result, getTimeStamp())
 
-            # error = checkNetworkClose()
-            # if error is not None:
-            #    return error
+        # on_cap(result, getTimeStamp())
 
-            while time.time() < frame_end - SLEEP_TIME:
-                time.sleep(SLEEP_TIME)
```

Comment:
TIL, `time.sleep()` timing precision _might_ have been fixed in all the major OSes and this hack might not be needed anymore

https://stackoverflow.com/a/15967564